rllib/algorithms/dqn/torch/dqn_rainbow_torch_learner.py

```diff
@@ -0,0 +1,238 @@
+                upper_bound.long(), self.config.num_atoms
+            )
+            ml_delta = probs_q_next_best * (upper_bound - b + floor_equal_ceil)
+            mu_delta = probs_q_next_best * (b - lower_bound)
+            ml_delta = torch.sum(lower_projection * ml_delta.unsqueeze(dim=-1), dim=1)
+            mu_delta = torch.sum(upper_projection * mu_delta.unsqueeze(dim=-1), dim=1)
+            # We do not want to propagate through the distributional targets.
+            m = (ml_delta + mu_delta).detach()
+
+            # Rainbow paper claims that using this form of cross-entropy loss for
+            # priority is robust and insensitive to 'prioritized_replay_alpha'.
+            # TODO (simon): Check this claim and get the exact location in the paper.
+            td_error = nn.CrossEntropyLoss(reduction="none")(q_logits_selected, m)
+            # Compute the weighted loss (importance sampling weights).
+            total_loss = torch.mean(batch["weights"] * td_error)
+        else:
+            # Masked all Q-values with terminated next states in the targets.
+            q_next_best_masked = (
+                1.0 - batch[SampleBatch.TERMINATEDS].float()
+            ) * q_next_best
+
+            # Compute the RHS of the Bellman equation.
+            # TODO (simon): Implement randomized n-step sampling.
+            # Detach this node from the computation graph as we do not want to
+            # backpropagate through the target network when optimizing the Q loss.
+            q_selected_target = (
+                batch[SampleBatch.REWARDS]
+                + self.config.gamma**self.config.n_step * q_next_best_masked
+            ).detach()
+
+            # Compute the TD error.
+            td_error = torch.abs(q_selected - q_selected_target)
+            # Compute the weighted loss (importance sampling weights).
+            total_loss = torch.mean(
+                batch["weights"]
+                * loss_fn(reduction="none")(q_selected, q_selected_target)
+            )
+
+        self.register_metrics(
+            module_id,
+            {
+                QF_LOSS_KEY: total_loss,
+                TD_ERROR_KEY: td_error,
+                QF_MEAN_KEY: torch.mean(q_selected),
+                QF_MAX_KEY: torch.max(q_selected),
+                QF_MIN_KEY: torch.min(q_selected),
+            },
+        )
+        # If we learn a Q-value distribution store the support and average
+        # probabilities.
+        if self.config.num_atoms > 1:
+            self.register_metrics(
+                module_id,
+                {
+                    ATOMS: z,
+                    QF_PROBS: torch.mean(fwd_out[QF_PROBS], dim=0),
+                },
+            )
+
+        return total_loss
+
+    @override(DQNRainbowLearner)
+    def _update_module_target_networks(
+        self, module_id: ModuleID, config: DQNConfig
+    ) -> None:
+        """Updates the target Q network(s) of a module.
```

Comment:
Same here, I don't think we need the docstring at all here. Move all the descriptions below into the super (abstract) method.

Reply:
To keep the code consistent let's implement the same design into the module definitions: 

- Parent class holds docstrings
- Child class has no docstrings for inherited methods